### ISSUE TITLE
Multi-copy orthologous genes

### ISSUE THREAD

**Issue by sumanthmutte** [JLSteenwyk/orthofisher#2]
Hi Jacob, 

Thanks for the nice tool. I was testing orthofisher using a sample dataset with a gene family that we are working with, Auxin Response Factors (ARF), in plants. There are 23 genes in this gene family in _Arabidopsis thaliana_. I have picked-up three proteins from this family and searched for it in the proteome using HMM that I have generated. As a result I expect to get (ideally) 23 genes or somewhere around or a bit less if the algorithm avoids/filters most remote homologs. But, I get only the three sequences that I gave as input to HMM in the results of orthfisher, indicating that the selection/filtering criteria for orthologs is too stringent. On the other hand, it correctly identifies the three existing orthologs in another proteome, _Marchantia polymorpha_. 

To reproduce the results, I have attached the files that I have used. Could you please check it and resolve this filtering so that multi-copy genes could be estimated properly? Or am I doing something differently?

Looking forward for your response!

Thanks and regards
Sumanth

[orthofisher_arf.zip](https://github.com/JLSteenwyk/orthofisher/files/5848627/orthofisher_arf.zip)

**Comment by JLSteenwyk** [comment]
Hi Sumanth,

I hope you are doing well and thank you for checking out orthofisher!

There are a number of parameters that may influence your results such as e-value threshold and the newly added _percent bitscore threshold_ (specified with the -b, --bitscore parameter). The current iteration of documentation describes the _percent bitscore threshold_ as the following:

`-b, –bitscore: A fraction threshold to specify the bitscore threshold for sequences to be considered similar. More specifically, if a hit has a bitscore less than the specified fraction, the gene will not be considered sufficiently similar to be considered putatively orthologous. Value must range from 0 to 1 (default: 0.85).`

Using your input files, I was able to identify 23 copies in ArabidopsisThaliana.fa using the following set of arguments `orthofisher -f fasta_arg.txt -m hmms.txt -b .15`; however, the 23 copies widely varied in their sequence similarity to the HMM with bitscores ranging from ~230-1200. Furthermore, I did identify 5 copies in MarchantiaPolymorpha.fa. Examination of output file orthofisher_output/hmmsearch_output/MarchantiaPolymorpha.fa_ARF.hmm suggests 5 is a reasonable number to identify. 

In summary, modifying the e-value threshold or percent bitscore threshold can heavily influence your results and are provided to enable researchers the flexibility to modify components of the pipeline to fit their research needs.

All the best,

Jacob

**Comment by sumanthmutte** [comment]
Hi Jacob,

Thank you for adding this new parameter. This is exactly what I was thinking to have an option to regulate (bit) score, so that the stringency for filtering can be controlled. I will test this soon. I hope this will be useful for others as well. 

Thank you once again for a nice tool and for your quick updates.

Best regards,
Sumanth

**Comment by JLSteenwyk** [comment]
That's great - thank you again Sumanth! 

I will go ahead and close this ticket.
